src/sfProject.ts

```diff
@@ -762,3 +713,26 @@ export class SfProject {
       .map(([key]) => key);
   }
 }
+
+/** differentiate between the Base PackageDir (path, maybe default) and the Packaging version (package and maybe a LOT of other fields) by whether is has the `package` property */
+export const isPackagingDirectory = (packageDir: PackageDir): packageDir is PackagePackageDir =>
+  isPackagingDir(packageDir);
+
+/** differentiate between the Base PackageDir (path, maybe default) and the Packaging version (package and maybe a LOT of other fields) by whether is has the `package` property */
+export const isNamedPackagingDirectory = (packageDir: NamedPackageDir): packageDir is NamedPackagingDir =>
+  isPackagingDir(packageDir);
+
+const isPackagingDir = (packageDir: PackageDir | NamedPackageDir): boolean =>
+  'package' in packageDir && typeof packageDir.package === 'string';
```

Comment:
isn't `typeof packageDir.package` always a string?

Reply:
it could be 'undefined'

Reply:
but it's typed as a required string? 
![image](https://github.com/forcedotcom/sfdx-core/assets/23369140/26221504-aa2d-4a22-91e1-e4698d8f4e18)


just checking `&& packageDir.package` should work too right? 🤷‍♂️